src/Mutex/Impl/LockRequest.cs

```diff
@@ -0,0 +1,3 @@
+namespace TemporalioSamples.Mutex.Impl;
+
+internal record LockRequest(string InitiatorId, string ResourceId, string AcquireLockSignalName, TimeSpan? Timeout = null);
```

Comment:
Can we remove the `ResourceId` from this since it's not a property of the request it's a property of the entire workflow? If the resource ID is needed from the workflow, can store on the workflow or derive from workflow ID.